src/Security/InheritedPermissionsExtension.php

```diff
@@ -14,6 +14,8 @@
  * @method ManyManyList<Member> EditorMembers()
  * @method ManyManyList<Group> ViewerGroups()
  * @method ManyManyList<Member> ViewerMembers()
+ *
+ * @extends DataExtension<DataObject&static>
```

Comment:
This could be applied in project code to any arbitrary model, so we shouldn't specify `File|SiteTree` here.

> Not sure how `&static` would be applicable here

`&static` says "And also it has the methods/properties available in this class". I've done that for any extension subclasses where we're referencing extension methods on the owner. Looks like I might have accidentally added it to this one as well, so I'll remove it from here.

But for example in `Hierarchy` we have `$this->owner->stageChildren()`, where `stageChildren()` is a public method defined in the `Hierarchy` extension itself.

Reply:
Removed `&static` from this one